editor/src/components/editor/actions/actions.spec.browser2.tsx

```diff
@@ -4054,97 +4067,22 @@ export var storyboard = (
-                data-testid='element-to-be-copied'
-                data-uid='div'
-              />
-            </div>
-          )
-          
-          export var storyboard = (
-            <Storyboard data-uid='sb'>
-              <Scene
-                style={{
-                  width: 419,
-                  height: 363,
-                  position: 'absolute',
-                  left: 212,
-                  top: 128,
-                }}
-                data-label='Playground'
-                data-uid='scene'
-              >
-                <App data-uid='app' />
-              </Scene>
-            </Storyboard>
-          )
-          `,
-          'await-first-dom-report',
-        )
-
-        await selectComponentsForTest(editor, [EP.fromString('sb/scene/app:root/div')])
-        await pressKey('c', { modifiers: cmdModifier })
-        await editor.getDispatchFollowUpActionsFinished()
-
-        const canvasRoot = editor.renderedDOM.getByTestId('canvas-root')
-
-        firePasteEvent(canvasRoot)
-
-        await clipboardMock.pasteDone
-        await editor.getDispatchFollowUpActionsFinished()
-
+        expect(renderResult.getEditorState().postActionInteractionSession).not.toBeNull()
         expect(
-          editor.getEditorState().editor.canvas.interactionSession?.interactionData.type,
-        ).toEqual('DISCRETE_REPARENT')
-
-        const canvasControlsLayer = editor.renderedDOM.getByTestId(CanvasControlsContainerID)
-        const originalElementBounds = editor.renderedDOM
-          .getAllByTestId('element-to-be-copied')
-          .at(0)!
-          .getBoundingClientRect()
-        await mouseMoveToPoint(canvasControlsLayer, {
-          x: originalElementBounds.x + 1,
-          y: originalElementBounds.y + 1,
-        })
-
-        await editor.getDispatchFollowUpActionsFinished()
-
-        expect(editor.getEditorState().editor.highlightedViews.map(EP.toString)).toEqual([
-          'sb/scene/app:root/div',
+          renderResult.getEditorState().derived.navigatorTargets.map(navigatorEntryToKey),
+        ).toEqual([
+          'regular-utopia-storyboard-uid/scene-aaa',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/bbb',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/bbb/ccc',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/bbb/ddd',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf', // <- the pasted element
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf/aab',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf/aad',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf/aal',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf/aal/aah',
+          'regular-utopia-storyboard-uid/scene-aaa/app-entity:aaa/aaf/aal/aaj',
```

Comment:
Great catch, the problem was that repeated paste is whitelisted only to work either when pasting into a flex layout or when both the element and its would-be sibling are `position: absolute` (we can improve on this in the future). I fixed the project (https://github.com/concrete-utopia/utopia/pull/3831/commits/561d1c95b4462cff9a7aa514cf42780042fe4f25) in the test so that all elements that affect the outcome of paste are `position: absolute`.

Reply:
Great catch, the problem was that repeated paste is whitelisted only to work either when pasting into a flex layout or when both the element and its would-be sibling are `position: absolute` (we can improve on this in the future). I fixed the project (https://github.com/concrete-utopia/utopia/pull/3831/commits/561d1c95b4462cff9a7aa514cf42780042fe4f25) in the test so that all elements that affect the outcome of paste are `position: absolute`.